traffic_portal/app/src/common/modules/form/cacheGroup/FormCacheGroupController.js

```diff
@@ -145,6 +145,17 @@ var FormCacheGroupController = function(cacheGroup, types, cacheGroups, $scope,
         $scope.fallbackSelected = '';
     };
 
+    $scope.updateForNewType = function() {
+        // removes Cache Group fallbacks if type has changed and is no longer EDGE_LOC
+        if (!$scope.isEdgeLoc(cacheGroup.typeId)) {
+            let currentFallbacksCount = cacheGroup.fallbacks.length;
+            for (var i = 0; i < currentFallbacksCount; i++) {
```

Comment:
Is it necessary to do this iteration? Could `cacheGroup.fallbacks` be set to an empty array, (or `null`/`undefined`, not sure which is appropriate tbh).

Reply:
I did it this way so that it adds the previously selected fallbacks back into the list of available options in the drop down (and in the correct order where they started).  Ultimately cacheGroup.fallbacks ends up as an empty array, but this also does the clean up to get them back as options

Reply:
I see